packages/components/buttons/icon-button/src/icon-button.tsx

```diff
@@ -18,6 +18,31 @@ import {
   getHoverStyles,
 } from './icon-button.styles';
 
+/**
+ * @deprecated Use '10' from `TSizes` instead.
+ */
+type TSmall = 'small';
+
+/**
+ * @deprecated Use '30' from `TSizes` instead.
+ */
+type TMedium = 'medium';
+
+/**
+ * @deprecated Use '40' from `TSizes` instead.
+ */
+type TBig = 'big';
```

Comment:
I was wondering if we actually need these types since this is a pattern we haven't used before.

Could you maybe explain what's the value of these against using those values in the `TLegacySizes` type?

```ts
type TLegacySizes = 'small` | `medium` | `big`;
``` 

I'm asking because they are not exported nor used within this file.

Thanks.

Reply:
Not completely aware of how everything is wired together yet, I had the slight hope that - from a consumer perspective - their editor won’t just indicate that “medium” is a deprecated value, but rather show the deprecation-message for the individual token. This way users could fix it right away in their editor instead of finding the docs online and a possible alternative value.

Reply:
Thanks for the explanation, Michael 🙇 

I think that's a great idea but I'm not sure how we can make it work.

I've just done a quick test and I don't seem to have the message you're looking for in the editor.
Do you know if that's actually possible?

![image](https://github.com/commercetools/ui-kit/assets/97907068/42d8827e-47d3-4c69-b5d4-2ee84baeb356)

Reply:
I’m tempted to say it is, but It might be just wishful thinking. I’ve seen stroke-lines when using react components, but am not sure anymore if it was the property or the value that got it :D

Reply:
My point of view is that we should introduce the changes for a reason that adds more value for developer experience or project maintainability.
In this case, I would love to see what you are proposing but if we can't achieve that goal, I would vote for going for the usual pattern so we don't introduce anything new that does not provide an actual value.

Let's see what others think about it.

Reply:
To be fair, I agree that it would have been be very nice to have it display a stroke for deprecated  properties on that granular level. But if that is not the case at the moment, it would be great to keep the initial workflow we already have since that doesn't seem to show up.

Reply:
addressed [here](https://github.com/commercetools/ui-kit/pull/2830/files#diff-864898bbe411df91fd117fb7208de606c7d979c399fabb1cf5dbe90933fcff94R24), [here](https://github.com/commercetools/ui-kit/pull/2830/files#diff-3a56ecdd8c0ec240bd13af3c79568baf5b6c619a5b6e6443dacdae064f24a7a4R21), [here](https://github.com/commercetools/ui-kit/pull/2830/files#diff-c7dc6615dcde1e6e98272ffe5ee052955de20e77af64f177c5f0c78b9792acc0R26) and [here](https://github.com/commercetools/ui-kit/pull/2830/files#diff-ab4d692c622bc07c621c7eb40977e7f7ab012f23e7d0674d4944849b880f0319L15). @CarlosCortizasCT